app/Http/Controllers/Leaderboard/LeaderboardController.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace App\Http\Controllers\Leaderboard;
+
+use App\Http\Controllers\Controller;
+use App\Models\User\User;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Redis;
+
+class LeaderboardController extends Controller
+{
+    /**
+     * Get the first paginated section of the global leaderboard
+     */
+    public function getPaginated ()
+    {
+        $perPage = 100;
+
+        $start = 0;
+        $end = 100;
+
+        // Get the current page
+        $currentPage = request('page') ?: 0;
+
+        if ($currentPage > 0)
+        {
+            // 101
+            $start = ($currentPage * $perPage) + 1;
+
+            // 200
+            $end = $end + ($currentPage * $perPage);
+        }
+
+        $userIds = Redis::zrevrange("xp.users", $start, $end);
+
+        $users = User::whereIn('id', $userIds)
+            ->select('id', 'show_name', 'show_username', 'name', 'username')
```

Comment:
todo - add socials

Reply:
Actually, we're better off just showing all users according to their XP. If they don't want their name shown, then we just show their name as Anonymous. That's similar to what we do in our existing country, state, and city leaderboards. This will make the code simpler because there's just too much duplication here.

Reply:
And will make it easier to include socials here.